include/gadget/buffer.h

```diff
@@ -18,6 +18,7 @@
 	} name SEC(".maps");				\
 	const void *gadget_map_tracer_##name __attribute__((unused));
 
+#ifndef GADGET_NO_RESERVE
```

Comment:
ref https://github.com/inspektor-gadget/inspektor-gadget/pull/2323#discussion_r1435262889

I'm still not convinced by this one, I don't like that we're adding an additional knob when it's not fully needed.

I see two options: 

1. Use `MAX_EVENT_SIZE` to control it, if it's defined then the gadget_reserve_buf / gadget_submit_buf will be defined, otherwise them won't be available. 
1. Split buffer.h in two header files, buffer-output.h (or similar) will contain the map definition and `gadget_output_buf`, buffer.h will include buffer-output.h, define the gadget_heap and other functions.

Reply:
> If every time I would need to #include a library I would need to #define something this would definitely make me refrain using this library.

The second option doesn't require this.

> The thing we should do here is to have the gadget_heap, if used, size set to struct eventin golang code with some magic to round it to the nearest power of 2.

Why to involve golang if this can be done directly in c? Also the heap doesn't have to be a power of 2 (as opposed to the ring size)

Reply:
> And then, we need to include two times...

The idea is to have buffer.h including buffer-ouput.h, hence from the user POV it's only needed to include one of the two files.

But I don't care too much about it TBH, if you still think the macro is the best approach let's go that way.

> Indeed for the power of pwo.
> By using golang no one has to compute offline the size of the struct as it would be done by go for users.

Makes sense.